File: Segundo/Monopoly/civitas_ruby (1)/proyecto_civitas_ruby/lib/controlador.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative 'civitas_juego'
require_relative 'operaciones_juego'
require_relative 'operacion_inmobiliaria'
require_relative 'gestiones_inmobiliarias'
require_relative 'salidas_carcel'
require_relative 'respuestas'

module Civitas
  class Controlador
    def initialize(juego,vista)
      @juego=juego
      @vista=vista
    end
    
    def juega()
      
      @vista.setCivitasJuego(@juego)
     
      while ( !@juego.finalDelJuego )
        operacion = @juego.siguientePaso
        @vista.actualizarVista
        @vista.pausa

        
        @vista.mostrarSiguienteOperacion(operacion)

        if (operacion != Civitas::OperacionesJuego::PASAR_TURNO)
          @vista.mostrarEventos
        end
        
       
        if (!@juego.finalDelJuego)
          case operacion
            when OperacionesJuego::COMPRAR
              if @vista.comprar == Respuestas::SI
                @juego.comprar
              end
              @juego.siguientePasoCompletado(operacion)
              
            
            when OperacionesJuego::GESTIONAR
              
              @vista.gestionar
              
              lista=[Civitas::GestionesInmobiliarias::VENDER,Civitas::GestionesInmobiliarias::HIPOTECAR,Civitas::GestionesInmobiliarias::CANCELAR_HIPOTECA,Civitas::GestionesInmobiliarias::CONSTRUIR_CASA,Civitas::GestionesInmobiliarias::CONSTRUIR_HOTEL,Civitas::GestionesInmobiliarias::TERMINAR]
              oi=Civitas::OperacionInmobiliaria.new(lista[@vista.iGestion], @vista.iPropiedad)
              
              case lista[@vista.iGestion]
                  when GestionesInmobiliarias::VENDER
                    if(@juego.getJugadorActual.get_nombre_propiedades!=nil)
                      @juego.vender(oi.getNumPropiedad)
                    end
                  when GestionesInmobiliarias::HIPOTECAR 
                    if(@juego.getJugadorActual.get_nombre_propiedades!=nil)
                      @juego.hipotecar(oi.getNumPropiedad)
                    end
                  when GestionesInmobiliarias::CANCELAR_HIPOTECA
                    if(@juego.getJugadorActual.get_nombre_propiedades!=nil)
                      @juego.cancelarHipoteca(oi.getNumPropiedad)
                    end
                  when GestionesInmobiliarias::CONSTRUIR_CASA
                    @juego.construirCasa(oi.getNumPropiedad)
                  when GestionesInmobiliarias::CONSTRUIR_HOTEL
                    @juego.construirHotel(oi.getNumPropiedad)
                  else
                    @juego.siguientePasoCompletado(operacion)
                
              end
              
          
          
          when   Civitas::OperacionesJuego::SALIR_CARCEL
              turn=@vista.salirCarcel
              case turn
                when SalidasCarcel::PAGANDO
                  @juego.salirCarcelPagando
                when SalidasCarcel::TIRANDO
                  @juego.salirCarcelTirando
              end
              @juego.siguientePasoCompletado(operacion)
              
            
            else
              list_aux=Array.new
              list_aux=@juego.ranking
              list_aux.reverse
              for i in 0...list_aux.size
                puts list_aux[i].nombre+"\n"
              end
            
          end
          
 
            
        end
          
          
          
      end
        
        
    end
      
  end
end

File: Segundo/Monopoly/civitas_java (1)/proyecto_civitas/src/GUI/Controlador.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package GUI;
//import civitas.Casilla;
import civitas.CivitasJuego;
//import civitas.Diario;
import civitas.GestionesInmobiliarias;
//import civitas.Jugador;
import civitas.OperacionesJuego;
import civitas.OperacionInmobiliaria;
import civitas.SalidasCarcel;
//import civitas.TituloPropiedad;

/**
 *
 * @author pc
 */
//import GUI.Respuestas;
//
//import static java.lang.System.exit;
//import java.util.ArrayList;
//import java.util.Arrays;
//import java.util.Scanner;
public class Controlador {
    private CivitasJuego juego;
    private CivitasView vista;
    
    
    public Controlador(CivitasJuego juego,CivitasView vista){
        this.juego=juego;
        this.vista=vista;
    }
    public void juega(){
        vista.setCivitasJuego(juego);
        while(!juego.finalDelJuego())
        {
            
            OperacionesJuego oper=juego.siguientePaso();
            
            vista.actualizarVista();
            
            
            vista.mostrarSiguienteOperacion(oper);
            
            if (oper != civitas.OperacionesJuego.PASARTURNO)
                vista.mostrarEventos();
            
            if(!juego.finalDelJuego())
            {
                switch(oper){
                    case COMPRAR: 
                        Respuestas opc_respuesta;
                        opc_respuesta=vista.comprar();
                        if(opc_respuesta == Respuestas.SI)
                            juego.comprar();

                        juego.siguientePasoCompletado(oper);
                        break;
                    case GESTIONAR:
                        if(juego.getJugadorActual().getPropiedades().isEmpty()==false)
                        {
                            vista.gestionar();
                            GestionesInmobiliarias c_gestion = GestionesInmobiliarias.values()[vista.getGestion()];

                            int i_prop = vista.getPropiedad();
                            OperacionInmobiliaria tarea=new OperacionInmobiliaria(c_gestion,i_prop);
                            switch(tarea.getGestion())
                            {
                                case VENDER:
                                    if(juego.getJugadorActual().getNombrePropiedades()!=null)
                                    {
                                        juego.vender(i_prop);
                                    }
                                    break;
                                case HIPOTECAR:
                                    if(juego.getJugadorActual().getNombrePropiedades()!=null)
                                        juego.hipotecar(i_prop);
                                    break;
                                case CANCELAR_HIPOTECA:
                                    if(juego.getJugadorActual().getNombrePropiedades()!=null)
                                        juego.cancelarHipoteca(i_prop);
                                    break;
                                case CONSTRUIR_CASA:
                                    juego.construirCasa(i_prop);
                                    break;
                                case CONSTRUIR_HOTEL:
                                    juego.construirHotel(i_prop);
                                    break;
                                default:
                                    juego.siguientePasoCompletado(oper);

                                break;
                            }
                        }
                        else
                            juego.siguientePasoCompletado(oper);
                        break;
                    case SALIRCARCEL:
                        SalidasCarcel salida=vista.salirCarcel();
                        switch(salida)
                        {
                            case PAGANDO:
                                juego.salirCarcelPagando();
                                break;
                            case TIRANDO:
                                juego.salirCarcelTirando();
                                break;
                        }
                    
                }
            }
            else
            {

                vista.actualizarVista();
            }
            
        }
    }
   
}
